### PR TITLE
Add KMS function

### PR DESCRIPTION
**Pull request by Zstro**

### CODE REVIEW

File: api/bucket-encryption-handlers.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	"net/http"
+)
+
+func (api ObjectAPIHandlers) PutBucketEncryption(w http.ResponseWriter, r *http.Request) {
+	ctx := getRequestContext(r)
+	logger := ctx.Logger
+
+	var credential common.Credential
+	var err error
+	switch signature.GetRequestAuthType(r) {
```
Comment (cannium) [inline api/bucket-encryption-handlers.go]: Maybe only 2 cases, v4/not v4?

File: plugins/kms_plugin.go [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+		return 400, nil, err
+	}
+	request.Header.Set("Content-Type", "application/json")
+	request.Header.Set("Accept", "application/json")
+	request.Header.Add("X-Auth-Token", token)
+
+	res, err := client.Do(request)
+	if err != nil {
+		return res.StatusCode, nil, err
+	}
```
Comment (cannium) [inline plugins/kms_plugin.go]: put `defer res.Body.Close()` here

File: plugins/kms_plugin.go [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+func (s *SDKMS) checkAndGetKMSKey() (bool, error) {
+	pageNumber := 1
+
+	data, err := s.ListKeys(pageNumber, MaxPageSize)
+	if err != nil {
+		return false, err
+	}
+	if data.TotalCount > 0 {
+		if data.TotalCount > MaxPageSize {
+			data, err = s.ListKeys(pageNumber+1, MaxPageSize)
```
Comment (cannium) [inline plugins/kms_plugin.go]: at most 2 pages?

Comment (Zstro) [reply plugins/kms_plugin.go]: The total count of key is at most 200

File: plugins/kms_plugin.go (hunk 2 above)
Comment (Zstro) [reply plugins/kms_plugin.go]: ok

File: api/bucket-encryption-handlers.go (hunk 1 above)
Comment (Zstro) [reply api/bucket-encryption-handlers.go]: ok

File: api/object-handlers.go [hunk 4]
```diff
@@ -1246,6 +1262,11 @@ func (api ObjectAPIHandlers) NewMultipartUploadHandler(w http.ResponseWriter, r
 			WriteErrorResponse(w, r, err)
 			return
 		}
+	} else if configuration, ok := checkBucketEncryption(bucketName, api.ObjectAPI); ok {
```
Comment (cxt90730) [inline api/object-handlers.go]: Could get bucketencryption by bucket info？

Comment (Zstro) [reply api/object-handlers.go]: I think some request have not such detailed bucket info

**Pull request merged**